src/cdk/a11y/focus-monitor.ts

```diff
@@ -58,8 +58,8 @@ export class FocusMonitor {
   /** The timeout id of the touch timeout, used to cancel timeout later. */
   private _touchTimeout: number;
 
-  /** Weak map of elements being monitored to their info. */
-  private _elementInfo = new WeakMap<Element, MonitoredElementInfo>();
+  /** Map of elements being monitored to their info. */
+  private _elementInfo = new Map<HTMLElement, MonitoredElementInfo>();
```

Comment:
Note: I would prefer to keep this as a `WeakMap`, but we need a `Map` in order to be able to iterate over the elements. Alternatively, if we make an assumption that all the monitored elements will be removed on destroy, we could remove only the global listeners.

Reply:
This is a tricky trade-off. We don't have any data to know for sure, but I expect _most_ apps will end up with `FocusMonitor` provided at the root, meaning that using `Map` could end up making elements remain in memory after they'd otherwise be removed. 

Do you think it would be common that monitored elements would remain while the module with the `FocusMonitor` would be unloaded?

Reply:
You're right, the `unlisten` would preserve the element reference already, so switching to a `Map` won't change anything